doc/sphinx/user-extensions/syntax-extensions.rst

```diff
@@ -1766,6 +1820,111 @@ The following errors apply to both string and numeral notations:
 
      .. todo note on "single qualified identifiers" https://github.com/coq/coq/pull/11718#discussion_r415076703
 
+.. example:: Number Notation for a non inductive type
+
+   The following example parses and prints :g:`prod unit (prod unit unit)` as :g:`3`.
+   The inductive :g:`I` will be used as :n:`@qualid__ind`.
```

Comment:
Is this a realistic example?  Realistic examples are better except if it makes the example a lot longer/more complex.  If appropriate, you could say that the example is a bit contrived.  Also, how about informally explaining the `prod` representation?  (Just curious: how do you represent zero and one?)

Reply:
I agree the example is a bit artificial, but I don't have any natural example in mind that would not be much longer.
Maybe @JasonGross can think of a less artificial example?

Reply:
What about an example that parses and prints `sum unit (sum unit ( .. (sum unit unit) .. ))` as the number of `unit`s, i.e., notations for ad-hoc non-empty finite types of natural-number size?